designs/PackageReference-Extern-Alias.md

```diff
@@ -0,0 +1,164 @@
+# Extern alias support with PackageReference
+
+* Status: In Review
+* Author(s): [Nikolche Kolev](https://github.com/nkolev92)
+* Issue: [4989](https://github.com/NuGet/Home/issues/4989) Extern alias support for NuGet package references
+* Type: Feature
+
+## Problem Background
+
+In the PackageReference world there is no way to add assembly specific metadata.
+Sometimes multiple packages might carry the same type, and in those cases the solution is to use the alias feature. However, there's no native way to do that in PackageReference.
+In comparison, in packages.config, one can add the extra metadata to the assembly included because all the assembly references are written in the project file. 
+
+## Who are the customers
+
+All PackageReference customers
+
+## Requirements
+
+* A feature or guidance to allow the customers to specify an alias for an assembly.
+
+## Goals
+
+* Define an approach to define alias for assemblies from a specific package.
+
+## Non-Goals
+
+* Define an approach for defining additional assembly metadata.
+* Define an approach for defining aliases or any other assembly metadata for packages brought in transitively.
+
+## Solution
+
+The problem of assembly metadata for assemblies is not unique unique to aliasing only, but it's definitely the most common one. 
+In PackageReference all customizations are on the PackageReference item itself. For example, `IncludeAssets`, `PrivateAssets`, `GeneratePathProperty` etc. 
+As such the proposed approach is the following: 
+
+```xml
+<ItemGroup>
+    <PackageReference Include="NuGet.Contoso.Library.Signed"  Version="1.26.0" Aliases="signed" />
+</ItemGroup>
+```
+
+* Applies to all the compile time assemblies.
+* Does not apply to transitive dependencies.
+* More than one alias is allowed in an equivalent fashion to [ProjectReference.Aliases](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.projectreference.aliases?). Specifically to define more than alias, specify `first,second`.
+
+The limitation of this approach is that it does not allow per assembly granularity. 
+However the intuitiveness of this approach as it's analogous to [ProjectReference.Aliases](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.projectreference.aliases?), and the fact that a large majority of public packages have only 1 assembly per package (per framework), were the deciding factor to adopt this approach.
+
+### Solution - Technical details
+
+In PackageReference, everything about packages is in the assets file, as such the changes are the following: 
+
+In the csproj the specification is: 
+
+```xml
+<ItemGroup>
+    <PackageReference Include="NuGet.Contoso.Library.Signed"  Version="5.6.0" Aliases="signed" />

```

Comment:
Call out that this is "comma, separated" so "signed, unsigned" should be allowed.

Reply:
Good point, will add it.